javaslang/src/test/java/javaslang/collection/BitSetTest.java

```diff
@@ -0,0 +1,389 @@
+package javaslang.collection;
+
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.Tuple3;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.IterableAssert;
+import org.assertj.core.api.ObjectAssert;
+import org.junit.Test;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+public class BitSetTest extends AbstractSortedSetTest {
+
+    private final static int MAX_BIT = 1_000_000;
+
+    private enum E {
+        V1, V2, V3
+    }
+
+    private static class Mapper<T> implements Serializable {
+
+        private static final long serialVersionUID = 1L;
+
+        private final java.util.Map<Integer, T> fromIntMap = new java.util.HashMap<>();
+        private final java.util.Map<T, Integer> toIntMap = new java.util.HashMap<>();
+        private int nextValue = 0;
+
+        synchronized T fromInt(Integer i) {
+            if (i < nextValue) {
+                return fromIntMap.get(i);
+            } else {
+                throw new RuntimeException();
+            }
+        }
+
+        synchronized Integer toInt(T value) {
+            Integer i = toIntMap.get(value);
+            if (i == null) {
+                i = nextValue++;
+                toIntMap.put(value, i);
+                fromIntMap.put(i, value);
+            }
+            return i;
+        }
+    }
+
+    @Override
+    protected <T> IterableAssert<T> assertThat(Iterable<T> actual) {
+        return new IterableAssert<T>(actual) {
+            @Override
+            @SuppressWarnings("unchecked")
+            public IterableAssert<T> isEqualTo(Object obj) {
+                if(obj instanceof BitSet || actual instanceof BitSet) {
+                    Assertions.assertThat(HashSet.ofAll(actual)).isEqualTo(HashSet.ofAll((Iterable<T>) obj));
+                } else {
+                    super.isEqualTo(obj);
+                }
+                return this;
+            }
+        };
+    }
+
+    @Override
+    protected <T> ObjectAssert<T> assertThat(T actual) {
+        return new ObjectAssert<T>(actual) {
+            @Override
+            public ObjectAssert<T> isEqualTo(Object expected) {
+                if (actual instanceof Tuple2) {
+                    final Tuple2<?, ?> t1 = (Tuple2<?, ?>) actual;
+                    final Tuple2<?, ?> t2 = (Tuple2<?, ?>) expected;
+                    assertThat((Iterable<?>) t1._1).isEqualTo(t2._1);
+                    assertThat((Iterable<?>) t1._2).isEqualTo(t2._2);
```

Comment:
No, it's usual behavior, but we need to to check that Tuple2&lt;TreeSet,TreeSet&gt; **corresponds** to Tuple2&lt;BitSet,BitSet&gt; i.e. has same elements set. It's used when we check result of methods that returns something else than <code>BitSet</code>, for example <code>unzip()</code>